packages/preview/invoice-maker/1.0.0/lib.typ

```diff
@@ -0,0 +1,448 @@
+        ),
+        GB: regex(
+          "^GB[a-zA-Z0-9]{2}\s?([a-zA-Z]{4}\s?){1}([0-9]{4}\s?){3}([0-9]{2})$"
+        ),
+      )
+
+    if country == none or not country in iban_regexes {
+      true
+    }
+    else {
+      iban.find(iban_regexes.at(country)) != none
+    }
+}
+
+#let TODO = box(
+  inset: (x: 0.5em),
+  outset: (y: 0.2em),
+  radius: 0.2em,
+  fill: rgb(255,180,170),
+)[
+  #text(
+    font: "Arial",
+    size: 8pt,
+    weight: 600,
+    fill: rgb(100,68,64)
+  )[TODO]
+]
+
+#let horizontalrule = [
+  #v(8mm)
+  #line(
+    start: (20%,0%),
+    end: (80%,0%),
+    stroke: 0.8pt + gray,
+  )
+  #v(8mm)
+]
+
+#let signature_line = line(length: 5cm, stroke: 0.4pt)
+
+#let endnote(num, contents) = [
+  #stack(dir: ltr, spacing: 3pt, super[#num], contents)
+]
+
+#let languages = (
+    en: (
+      country: "GB",
+      recipient: "Recipient",
+      biller: "Biller",
+      invoice: "Invoice",
+      cancellation_invoice: "Cancellation Invoice",
+      cancellation_notice: (id, issuing_date) => [
+        As agreed, you will receive a credit note
+        for the invoice *#id* dated *#issuing_date*.
+      ],
+      invoice_id: "Invoice ID",
+      issuing_date: "Issuing Date",
+      delivery_date: "Delivery Date",
+      items: "Items",
+      closing: "Thank you for the good cooperation!",
+      number: "№",
+      date: "Date",
+      description: "Description",
+      duration: "Duration",
+      quantity: "Quantity",
+      price: "Price",
+      total_time: "Total working time",
+      subtotal: "Subtotal",
+      vat: "VAT of",
+      reverse_charge: "Reverse Charge",
+      total: "Total",
+      due_text: val =>
+        [Please transfer the money onto following bank account due to *#val*:],
+      owner: "Owner",
+      iban: "IBAN",
+    ),
+    de: (
+      country: "DE",
+      recipient: "Empfänger",
+      biller: "Aussteller",
+      invoice: "Rechnung",
+      cancellation_invoice: "Stornorechnung",
+      cancellation_notice: (id, issuing_date) => [
+        Vereinbarungsgemäß erhalten Sie hiermit eine Gutschrift
+        zur Rechnung *#id* vom *#issuing_date*.
+      ],
+      invoice_id: "Rechnungsnummer",
+      issuing_date: "Ausstellungsdatum",
+      delivery_date: "Lieferdatum",
+      items: "Leistungen",
+      closing: "Vielen Dank für die gute Zusammenarbeit!",
+      number: "Nr",
+      date: "Datum",
+      description: "Beschreibung",
+      duration: "Dauer",
+      quantity: "Menge",
+      price: "Preis",
+      total_time: "Gesamtarbeitszeit",
+      subtotal: "Zwischensumme",
+      vat: "Umsatzsteuer von",
+      reverse_charge: "Steuerschuldnerschaft des\nLeistungsempfängers",
+      total: "Gesamt",
+      due_text: val =>
+        [Bitte überweise den Betrag bis *#val* auf folgendes Konto:],
+      owner: "Inhaber",
+      iban: "IBAN",
+    ),
+  )
+
+#let invoice(
+  language: "en",
+  country: none,
+  title: none,
+  banner_image: none,
```

Comment:
I would recommend kebab-case as the canonical Typst case.

Reply:
Ok, will change it everywhere!
But honestly, I think it's not an ideal choice. Kebab-case isn't selectable with a double-click (at least not per default), which makes working with more cumbersome. Also interesting choice since you're coming from Rust. 😛

Reply:
There is no great case choice for Typst unfortunately. All have downsides. Snake case in particular has overlap with emphasis syntax and math mode (though kebab case also struggles with the latter). Kebab-case imo looks nicer for CSS-style set rules and some consider it more friendly to beginners. There were a bunch of discussion about this on the Discord at various times.

Reply:
Ok, I see.
Also, I realized double click behavior can be customized: https://stackoverflow.com/questions/34781471/vscode-is-there-a-way-to-customise-double-click-select-behaviour

Didn't have time to try it out yet, but if somebody figures out a good configuration for Typst, that would be a good addition to the docs!